scripts/getChangelog.py

```diff
@@ -1,90 +1,570 @@
+          unique_pr_numbers.add(pr_number)
+
+  # Create a list of dictionaries with PR metadata for in CSV file
+  animation_state['text'] = "Get PR metadata"
+  ## Combined GraphQL call: get relevant PRs and the issues they close
+  query = f'''
+  query ($owner: String!, $repo: String!) {{
+    repository (name: $repo, owner: $owner) {{
+  '''
+
+  for n, pr_number in enumerate(unique_pr_numbers):
+      query += f'''
+        pr{n}:pullRequest (number: {pr_number}) {{
+          url
+          title
+          author {{
+            login
+          }}
+          mergedAt
+          labels (first: 10) {{
+            totalCount
+            nodes {{
+              id
+              name
+            }}
+          }}
+          closingIssuesReferences(first: 10) {{
+            totalCount
+            nodes {{
+              number
+              title
+              labels(first: 10) {{
+                totalCount
+                nodes {{
+                  id
+                  name
+                }}
+              }}
+            }}
+          }}
+        }}
+        '''
+
+  query += f'''
+    }}
+  }}
+  '''
+  # Variable definition for GraphQL Explorer:
+  # {"prnumber": 7053, "owner": "AntennaPod", "repo": "AntennaPod"}
+
+  ## Submit call
+  variables = {"owner": owner, "repo": repo}
+  result = client.execute(query, variables)
+  data = json.loads(result)
+      
+  ## Parse response and save PR data
+  animation_state['text'] = "Parse PR metadata"
+  prs_for_csv = []
+
+  for n, pr_number in enumerate(unique_pr_numbers):
+      prdata = data['data']['repository'][f'pr{n}']
+      
+      # Create string with related issues
+      maximum_hit = False
+      related_issue_numbers = [relatedIssue['number'] for relatedIssue in prdata['closingIssuesReferences']['nodes']]
+      related_issues_string = ', '.join(map(str, related_issue_numbers))
+      if prdata['closingIssuesReferences']['totalCount'] > 10:
+        related_issues_string += " and more"
+      
+      # Create string with labels of the PR and its associated issues
+      unique_labels = set()
+      maximum_hit = False
+      if prdata['labels']['totalCount'] > 10:
+        maximum_hit = True
+      for label in prdata['labels']['nodes']:
+        unique_labels.add(label['name'])
+      for relatedIssue in prdata['closingIssuesReferences']['nodes']:
+        if relatedIssue['labels']['totalCount'] > 10:
+          maximum_hit = True
+        for label in relatedIssue['labels']['nodes']:
+          unique_labels.add(label['name'])
+          
+      unique_labels_list = list(unique_labels)
+      unique_labels_string = ', '.join(unique_labels_list)
+      if maximum_hit:
+        unique_labels_string += " and more (probably)"
+      
+      # Create string with issue & PR number(s) that need review replies
+      numbers = []
+      maximum_hit = False
+      if any(label['id'] == 'LA_kwDOAFAGHc8AAAABoGK6aw' for label in prdata['labels']['nodes']):
```

Comment:
Can this be done based on the label name instead of label id? That would make it easier to read

Reply:
It could, but if we tweak the label text, it would break the script. Maybe I should change replace it with a variable defined in the variable list, so it can more easily changed by the user.